queries.sql

```diff
@@ -732,8 +732,209 @@ FROM
+-- ------------------+---------
+--  Jack Harkness    | Digimon
+--  Stephanie Mendez | Pokemon
+--  Stephanie Mendez | Digimon
+--  William Tatcher  | Pokemon
+-- (4 rows)
+----------------------------------------------------------------------------------------
+/* List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020. */
+SELECT
+  visits.date_of_visit AS visited_date,
+  animals.name AS visitor
+FROM
+  visits
+  JOIN animals ON animal_id = animals.species_id
+  JOIN vets ON vet_id = vets.id
+WHERE
+  vets.name = 'Stephanie Mendez'
+  AND date_of_visit between DATE '2020-04-01'
+  AND '2020-08-30'
+ORDER BY
+  date_of_visit DESC;
+
+--  visited_date |  visitor
+-- --------------+------------
+--  2020-07-22   | Boarmon
+--  2020-07-22   | Angemon
+--  2020-07-22   | Devimon
+--  2020-07-22   | Plantmon
+--  2020-07-22   | Gabumon
+--  2020-07-22   | Agumon
+--  2020-05-24   | Blossom
+--  2020-05-24   | Squirtle
+--  2020-05-24   | Charmander
+--  2020-05-24   | Pikachu
+-- (10 rows)
+------------------------------------------------------------------------------
+/* What animal has the most visits to vets? */
+SELECT
+  species.name AS animal,
+  count(*) AS most_VISITS
+FROM
+  visits
+  JOIN species ON animal_id = species.id
+GROUP BY
+  species.name
+ORDER BY
+  most_VISITS DESC
+LIMIT
+  1;
+
+--  animal  | most_visits
+-- ---------+-------------
+--  Digimon |          13
+-- (1 row)
+------------------------------------------------------------------------------
+/*Who was Maisy Smith's first visit?*/
+SELECT
+  species.name AS visited,
+  vets.name AS vet
+FROM
+  visits
+  JOIN species ON animal_id = species.id
+  JOIN vets ON vet_id = vets.id
+WHERE
+  vets.name = 'Maisy Smith'
+ORDER BY
+  date_of_visit ASC
+LIMIT
+  1;
+
+--  visited |     vet
+-- ---------+-------------
+--  Digimon | Maisy Smith
+-- (1 row)
+------------------------------------------------------------------------------------
+/* Details for most recent visit: animal information, vet information, and date of visit */
+SELECT
+  vets.name AS vet_name,
+  species.name AS specialist_of,
+  animals.name AS animal_name,
+  species.name AS animal_type,
+  visits.date_of_visit
+from
+  visits
+  JOIN animals ON animal_id = animals.species_id
+  JOIN vets ON vet_id = vets.id
+  JOIN species ON animal_id = species.id
+ORDER BY
+  date_of_visit ASC
+LIMIT
+  1;
+
+--   vet_name   | specialist_of | animal_name | animal_type | date_of_visit
+-- -------------+---------------+-------------+-------------+---------------
+--  Maisy Smith | Digimon       | Agumon      | Digimon     | 2019-01-24
+-- (1 row)
+--------------------------------------------------------------------------------------
+/* How many visits were with a vet that did not specialize in that animal's species? */
+SELECT
+  count(*)
+FROM
+  visits
+  JOIN specializations ON visits.vet_id = specializations.vet_id
+WHERE
+  animal_id != species_id;
```

Comment:
- [Required] I think to get _How many times visits were with a vet that did not specialize in that animal's species?_  you should have joins between the `animals`, `visits`, `vets`, and the `species` table. Currently, your query returns an incorrect result because it only has joins between two tables, please see the query below as an example of how this can be achieved.

```SQL
  SELECT count(*)
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE animals.species_id NOT IN (SELECT species_id FROM specializations WHERE vet_id = vets.id)

```